Microsoft.Toolkit.Uwp.Notifications/Toasts/Compat/ToastNotificationManagerCompat.cs

```diff
@@ -0,0 +1,582 @@
+            }
+
+            // Make sure their arguments are set correctly
+            var argumentsNode = comExeServerNode.Attributes.GetNamedItem("Arguments");
+            if (argumentsNode == null || argumentsNode.Value != "-ToastActivated")
+            {
+                throw new InvalidOperationException("Your app manifest's comServer extension for toast activation must have its Arguments set exactly to \"-ToastActivated\"");
+            }
+
+            return clsid;
+        }
+
+        /// <summary>
+        /// From https://stackoverflow.com/a/41622689/1454643
+        /// Generates Guid based on String. Key assumption for this algorithm is that name is unique (across where it it's being used)
+        /// and if name byte length is less than 16 - it will be fetched directly into guid, if over 16 bytes - then we compute sha-1
+        /// hash from string and then pass it to guid.
+        /// </summary>
+        /// <param name="name">Unique name which is unique across where this guid will be used.</param>
+        /// <returns>For example "706C7567-696E-7300-0000-000000000000" for "plugins"</returns>
+        private static string GenerateGuid(string name)
+        {
+            byte[] buf = Encoding.UTF8.GetBytes(name);
+            byte[] guid = new byte[16];
+            if (buf.Length < 16)
+            {
+                Array.Copy(buf, guid, buf.Length);
+            }
+            else
+            {
+                using (SHA1 sha1 = SHA1.Create())
+                {
+                    byte[] hash = sha1.ComputeHash(buf);
+
+                    // Hash is 20 bytes, but we need 16. We loose some of "uniqueness", but I doubt it will be fatal
+                    Array.Copy(hash, guid, 16);
+                }
+            }
+
+            // Don't use Guid constructor, it tends to swap bytes. We want to preserve original string as hex dump.
+            string guidS = $"{guid[0]:X2}{guid[1]:X2}{guid[2]:X2}{guid[3]:X2}-{guid[4]:X2}{guid[5]:X2}-{guid[6]:X2}{guid[7]:X2}-{guid[8]:X2}{guid[9]:X2}-{guid[10]:X2}{guid[11]:X2}{guid[12]:X2}{guid[13]:X2}{guid[14]:X2}{guid[15]:X2}";
+
+            return guidS;
+        }
+
+        private static void CreateAndRegisterActivator()
+        {
+            var activatorType = CreateActivatorType(_aumid);
+            RegisterActivator(activatorType);
+            _registeredOnActivated = true;
+        }
+
+        private static void RegisterActivator(Type activatorType)
+        {
+            if (!DesktopBridgeHelpers.IsRunningAsUwp())
+            {
+                string exePath = Process.GetCurrentProcess().MainModule.FileName;
+                RegisterComServer(activatorType, exePath);
+            }
+
+            // Big thanks to FrecherxDachs for figuring out the following code which works in .NET Core 3: https://github.com/FrecherxDachs/UwpNotificationNetCoreTest
```

Comment:
This is also under MIT, but I think we need to add it somewhere, don't we @michael-hawker ?